assets/src/design-system/components/switch/index.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * Internal dependencies
+ */
+import { themeHelpers } from '../../theme';
+import { Text } from '../typography';
+import { useKeyDownEffect } from '../keyboard';
+
+const SWITCH_HEIGHT = 32;
+
+const VisuallyHiddenRadioGroupLabel = styled.h4`
+  ${themeHelpers.visuallyHidden};
+`;
+
+const SwitchContainer = styled.div`
+  position: relative;
+  height: ${SWITCH_HEIGHT}px;
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  appearance: none;
+  background: ${({ theme }) => theme.colors.divider.secondary};
+  border-radius: 100px;
+  color: ${({ theme }) => theme.colors.fg.primary};
+`;
+
+const HiddenRadioButton = styled.input.attrs({ type: 'radio' })`
+  position: absolute;
+  height: 0;
+  width: 0;
+  padding: 0;
+  min-width: unset;
+  appearance: none;
+  border: 0;
+  box-shadow: none;
+  margin: -1px;
+  outline: none;
+  overflow: hidden;
+`;
+
+const RadioButtonLabel = styled(Text).attrs({ forwardedAs: 'label' })`
+  flex: 1;
+  width: 50%;
+  padding: 0px 6px;
+  z-index: 1;
+  color: ${({ isActive, theme }) =>
+    isActive ? theme.colors.inverted.fg.primary : theme.colors.fg.secondary};
+  cursor: pointer;
+  overflow: hidden;
+  text-align: center;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  transition: color 0.15s ease-out;
+
+  ${({ disabled }) =>
+    disabled &&
+    css`
+      cursor: default;
+      opacity: 0.3;
+    `}
+
+  /* add focus styling on the slider when the hidden input is focused */
+  :focus-within ~ span {
+    ${themeHelpers.focusCSS};
+  }
+`;
+
+const SlidingButton = styled.span`
+  position: absolute;
+  display: block;
+  top: 0;
+  left: 0;
+  width: 50%;
+  height: ${SWITCH_HEIGHT}px;
+  border-radius: 100px;
+  background-color: ${({ theme }) => theme.colors.interactiveBg.primaryNormal};
+  transition: left 0.15s ease-out;
+  z-index: 0;
+
+  ${({ hasOffset }) => hasOffset && `left: 50%`}
+`;
+
+/**
+ * A controlled radio group that looks like a switch.
+ *
+ * Keyboard navigation:
+ * - [Tab] can be used to navigate between radio groups
+ * - [Enter], [Space], [ArrowLeft], [ArrowRight] [ArrowUp], and [ArrowDown] can be used to select radio buttons
+ *
+ * @param {Object} props The props
+ * @param {string} props.className className that is applied to the outer container
+ * @param {string} props.disabled whether the radio buttons are disabled
+ * @param {string} props.groupLabel the label for the radio group. Used for accessibility
+ * @param {string} props.id unique id used for accessibility
+ * @param {string} props.name name of the radio group
+ * @param {string} props.offLabel label for the 'off' radio button
+ * @param {Function} props.onChange change event handler
+ * @param {string} props.onLabel label for the 'on' radio button
+ * @param {boolean} props.value the value of the radio group
+ * @return {Object} The radio button
+ */
+export const Switch = forwardRef(function (
+  {
+    className = '',
```

Comment:
didn't add a `mousetrap` class here because we may want to use this apart from having that class.